src/integration_tests/datasets/dataset_endpoints_integration_test.py

```diff
@@ -0,0 +1,107 @@
+import pytest
+from unittest import TestCase
+from src.app.config.config_factory import config
+from src.test_data.dataset_test_data import dataset_metadata_collection_endpoints
+from src.test_data.dataset_test_data import dataset_unit_data_collection_endpoints
+from repositories.firebase.firebase_loader import firebase_loader
+from src.integration_tests.helpers.integration_helpers import (
+    cleanup,
+    generate_headers,
+    setup_session,
+    inject_wait_time,
+)
+from src.integration_tests.helpers.firestore_helpers import upload_dataset
+from google.cloud import firestore
+
+class DatasetEndpointsIntegrationTest(TestCase):
+    """
+    Integration tests for the Dataset Endpoints.
+
+    This test covers fetching metdata and unit data from firestore,
+    and checking that dataset metadata and unit data is handled correctly.
+    """
+    session = None
+    headers = None
+    firestore_client = None
+    
+    @classmethod
+    def setup_class(self) -> None:
+        cleanup()
+        inject_wait_time(3) # Inject wait time to allow resources properly set up
+        self.session = setup_session()
+        self.headers = generate_headers()
+        self.firestore_client = firestore.Client(project=config.PROJECT_ID, database=f"{config.PROJECT_ID}-sds")
+        self.dataset = upload_dataset(self.firestore_client, dataset_metadata_collection_endpoints, dataset_unit_data_collection_endpoints)
+        
+
+    @classmethod
+    def teardown_class(self) -> None:
+        cleanup()
+        inject_wait_time(3) # Inject wait time to allow all message to be processed
+
+    @pytest.mark.order(1)
+    def test_get_dataset_metadata_collection(self):
+        """
+        Test retriving dataset metadata.
+
+        - Sends a GET request to retrieve metadata for a dataset
+        - Asserts the metadata retrieved matches the expected structure.
+        """
+
+        response = self.session.get(
+            f"{config.API_URL}/v1/dataset_metadata?survey_id=test_survey_id_1&period_id=test_period_id_1",
```

Comment:
Should reference survey_id and period_Id from the test_data used. This is to ensure the assertion is always true when the test is referring to a particular dataset, instead of being determined by the hardcoded query value which will invalidate the assertion if the data is changed.